src/distribution.cpp

```diff
@@ -26,30 +27,69 @@ Discrete::Discrete(pugi::xml_node node)
   auto params = get_node_array<double>(node, "parameters");
 
   std::size_t n = params.size();
-  std::copy(params.begin(), params.begin() + n / 2, std::back_inserter(x_));
-  std::copy(params.begin() + n / 2, params.end(), std::back_inserter(p_));
+  double x[n / 2], p[n / 2];
 
-  normalize();
+  std::copy(params.begin(), params.begin() + n / 2, x);
+  std::copy(params.begin() + n / 2, params.end(), p);
+
+  new (this) Discrete(x, p, n / 2);
 }
 
 Discrete::Discrete(const double* x, const double* p, int n)
-  : x_ {x, x + n}, p_ {p, p + n}
+  : x_ {x, x + n}, prob_ {p, p + n}
 {
   normalize();
```

Comment:
The normalization should be part of the constructor rather than a separate method. It doesn't make sense to call `normalize()` after the object is already constructed.